dspace-server-webapp/src/test/java/org/dspace/app/rest/RequestItemRepositoryIT.java

```diff
@@ -0,0 +1,368 @@
+     * @throws java.io.IOException passed through.
+     */
+    @Test
+    public void testCreateAndReturnAuthenticated()
+            throws SQLException, AuthorizeException, IOException, Exception {
+        System.out.println("createAndReturn");
+
+        context.turnOffAuthorisationSystem();
+
+        // Create some necessary objects.
+        Collection col = CollectionBuilder.createCollection(context,
+                parentCommunity).build();
+        Item item = ItemBuilder.createItem(context, col).build();
+        InputStream is = new ByteArrayInputStream(new byte[0]);
+        Bitstream bitstream = BitstreamBuilder.createBitstream(context, item, is)
+                .withName("/dev/null")
+                .withMimeType("text/plain")
+                .build();
+
+        // Fake up a request in REST form.
+        RequestItemRest rir = new RequestItemRest();
+        rir.setBitstreamId(bitstream.getID().toString());
+        rir.setItemId(item.getID().toString());
+        rir.setRequestEmail(RequestItemBuilder.REQ_EMAIL);
+        rir.setRequestMessage(RequestItemBuilder.REQ_MESSAGE);
+        rir.setRequestName(RequestItemBuilder.REQ_NAME);
+        rir.setAllfiles(false);
+
+        // Create it and see if it was created correctly.
+        ObjectMapper mapper = new ObjectMapper();
+        String authToken = getAuthToken(admin.getEmail(), password);
+        MvcResult mvcResult = getClient(authToken)
+                .perform(post(URI_ROOT)
+                        .content(mapper.writeValueAsBytes(rir))
+                        .contentType(contentType))
+                .andExpect(status().isCreated())
+                .andExpect(content().contentType(contentType))
+                .andExpect(jsonPath("$", Matchers.allOf(
+                        hasJsonPath("$.id", not(is(emptyOrNullString()))),
+                        hasJsonPath("$.type", is(RequestItemRest.NAME)),
+                        hasJsonPath("$.token", not(is(emptyOrNullString()))),
+                        hasJsonPath("$.requestEmail", is(RequestItemBuilder.REQ_EMAIL)),
+                        hasJsonPath("$.requestMessage", is(RequestItemBuilder.REQ_MESSAGE)),
+                        hasJsonPath("$.requestName", is(RequestItemBuilder.REQ_NAME)),
+                        hasJsonPath("$.allfiles", is(false)),
+                        hasJsonPath("$.requestDate", not(is(emptyOrNullString()))), // TODO should be an ISO datetime
+                        hasJsonPath("$._links.self.href", not(is(emptyOrNullString())))
+                )))
+                .andReturn();
+
+        // Clean up the created request.
+        String content = mvcResult.getResponse().getContentAsString();
+        Map<String,Object> map = mapper.readValue(content, Map.class);
+        String requestToken = String.valueOf(map.get("token"));
+        RequestItem ri = requestItemService.findByToken(context, requestToken);
+        requestItemService.delete(context, ri);
```

Comment:
The cleanup can also be simplified if you use an `andDo()` obviously.

I'd also recommend moving the deletion logic to a static method like `RequestItemBuilder.deleteRequestItem([requestToken])`, as this would better mirror other tests.  See for example how ItemBuilder has an `ItemBuilder.deleteItem([uuid])` method for deleting a specific item.

Reply:
Done.